### ISSUE TITLE
fix deprecation (since Xcode 7): `xcodebuild: WARNING: -exportArchive without -exportOptionsPlist is deprecated`

### ISSUE THREAD

**Issue by viktorbenei** [bitrise-io/steps-xcode-archive#7]
it's in the logs: `xcodebuild: WARNING: -exportArchive without -exportOptionsPlist is deprecated`

**Comment by Haud** [comment]
I believe this exact issue was causing my app to crash on launch using this script. Long story short, I think not specifying an -exportOptionsPlist using Xcode 7 when distributing via enterprise certs causes the bundle to omit a specific cert file that causes the app to crash on launch. I tried using a custom script that uses `xcodebuild --exportArchive` with an `exportOptionsPlist` in my app's directory, but I ran into some other issues with some other gems or something -- an error that I was unable to debug properly (maybe someone else can figure it out).

Ultimately, after a lot of trial and error, I resorted to using [shenzhen](https://github.com/nomad/shenzhen) (which is how we did builds previously on Jenkins and uses `xcrun PackageApplication` which seems to work fine) and getting it to export the proper environment variables to work with hockey scripts, etc.

Here's my script:

```
#!/bin/bash

gem install shenzhen

ipa build --scheme ${BITRISE_SCHEME} \
    --config ${BITRISE_CONFIGURATION} \
    --destination ${BITRISE_DEPLOY_DIR}

            
ipa_path=$(find ${BITRISE_DEPLOY_DIR} -name "*.ipa" | head -1)
dsym_path=$(find ${BITRISE_DEPLOY_DIR} -name "*.dSYM.zip" | head -1)

envman add --key BITRISE_IPA_PATH --value "${ipa_path}"
envman add --key BITRISE_DSYM_PATH --value "${dsym_path}”
```

You will have to modify the `ipa build` portion of the script to your liking; I specified a `scheme` and added a new app environment variable `${BITWISE_CONFIGURATION}` which specifies the build configuration to use when archiving. The rest of the script should be fine: it finds the ipa and dsym path and then sets the environment variables (just make sure not to change the destination path or, if you do, to modify the `ipa_path` and `dsym_path` to reference that new directory.

Hopefully this helps someone else running into the same issue. In the meantime, it would be great if BitRise were to update this script to no longer use the deprecated `exportFormat ipa` command and instead use `-exportOptionsPlist`.

**Comment by viktorbenei** [comment]
Thanks for reporting @Haud - we'll fix this ASAP. We're a bit resource limited because of the nearing Android support, but this issue is at the top of our list!

**Comment by Haud** [comment]
Awesome @viktorbenei .

For reference, here was the error I was getting when running `xcodebuild --exportArchive` with an `exportOptionsPlist` referenced in my project directory:

```

2015-10-17 12:05:23.150 xcodebuild[4552:14371] [MT] IDEDistribution: -[IDEDistributionLogging _createLoggingBundleAtPath:]: Created bundle at path '/var/folders/4f/945ywsh506g31f58fpcbg0840000gn/T/Scotts_2015-10-17_12-05-23.148.xcdistributionlogs'.
2015-10-17 12:05:25.289 xcodebuild[4552:14371] [MT] IDEDistribution: Step failed: : Error Domain=IDEDistributionErrorDomain Code=14 "No applicable devices found." UserInfo=0x7f84340f1be0 {NSLocalizedDescription=No applicable devices found.}
error: exportArchive: No applicable devices found.
Error Domain=IDEDistributionErrorDomain Code=14 "No applicable devices found." UserInfo=0x7f84340f1be0 {NSLocalizedDescription=No applicable devices found.}
** EXPORT FAILED **
```

For reference, running the script locally worked fine for me, so I can confirm there were no issues with my plist.

**Comment by viktorbenei** [comment]
Fixed in v1.3.0 https://github.com/bitrise-io/steps-xcode-archive/releases/tag/1.3.0 - thank you for creating the issue!
